### PR TITLE
Bugfix: Fixes document filename date off by 1 issue

### LINKED ISSUE #926: [BUG] document_renamer dates in filename off by one day
### Description

After running the `document_renamer`, all dates in the filename are off by one day. E.g. a document with the date 11.05.2022 will have 10.05.2022 in the filename after running the exporter.

This is related to #267, however since the original issue was regarding the `document_exporter` which seems to be fixed in the new version v1.7.1, I decided to open a new issue.

### Steps to reproduce

1. (Since this is probably a time zone issue: Set Timezone to Europe/Berlin)
2. Set `PAPERLESS_FILENAME_FORMAT` to something that contains `{created_day}`
3. Modify some details of the document (e.g. the tile) using the web ui
4. The filename of the pdf will contain the correct document day
5. Now run the `document_renamer`
6. The document day in the file name will be off by one day

### Webserver logs

_No response_

### Paperless-ngx version

1.7.1

### Host OS

x86_64

### Installation method

Docker

### Browser

_No response_

### Configuration changes

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by stumpylog**
## Proposed change

This PR fixes it so the filename of a document, both during a rename, save or export, will use the local time zone, instead of UTC time zone, which can, depending on the local time zone, result in an off by one in the date.

Dates being very hard, here's an example:
Document creation date: `2012-12-12 00:00:00+01:00` (Europe/Berlin)
Saved in database as `2012-12-11 23:00:00` (UTC)

Previously, the UTC time would be used to build an ISO format date like `2012-12-11` during document renaming, exporting and generally saving a document.  But then a user looks at the date in the web ui, sees `2012-12-12` (or the equivalent), and wonders why the date is off.

So all this really does it take the UTC date and convert back to local time zone, before using it to build a document filename.  I confirmed the filename is now matching local time zone for the initial consume, after changing the date in the web ui and when exported.

Fixes #926

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
Nice. So this respects `PAPERLESS_TIME_ZONE` or falls back to the system tz (in docker `TZ`) right? Is that what this does: ?https://github.com/paperless-ngx/paperless-ngx/blob/9a1bd9637cbef0e7318a3c61cae14805c6510cea/src/paperless/settings.py#L354

**Comment by stumpylog** [comment]
PAPERLESS_TIME_ZONE sets the Django [TIME_ZONE](https://docs.djangoproject.com/en/4.0/ref/settings/#std:setting-TIME_ZONE) setting.  So that's what all conversions to a local date or getting of a date through `django.utils.timezone` will use to create time zone aware dates.  Since these changes use the `django.utils.timezone` versions, they respect the set time zone.

File: src/documents/file_handling.py [hunk 1]
```diff
@@ -158,18 +159,22 @@ def generate_filename(doc, counter=0, append_gpg=True, archive_filename=False):
             else:
                 asn = "none"
 
+            # Convert UTC database date to localized date
+            local_added = timezone.localdate(doc.added)
+            local_created = timezone.localdate(doc.created)
```
Comment (lippoliv) [inline src/documents/file_handling.py:164]: The code before dealt with the situation, that `doc.created` probably doesn't exist, same for `doc.added`. How do you now deal with that situation?

File: src/documents/tests/test_document_model.py [hunk 2]
```diff
@@ -56,34 +56,43 @@ def test_file_name(self):
...
+        local_create_date = timezone.datetime(
+            2020,
+            12,
+            25,
+            tzinfo=zoneinfo.ZoneInfo("Europe/Berlin"),
+        )
+
+        utc_create_date = local_create_date.astimezone(zoneinfo.ZoneInfo("UTC"))
+
+        self.assertEqual(utc_create_date.date().day, 24)
```
Comment (lippoliv) [inline src/documents/tests/test_document_model.py:80]: Can you move this to the other assert? Not so sure if the project follows [GivenWhenThen](https://martinfowler.com/bliki/GivenWhenThen.html) officially, but I think esp. with growing test code base DEVs can review tests more easy.

**Review by lippoliv — CHANGES_REQUESTED** [review]
@stumpylog Overall looks good. But beside some code-style / variables, what is the change compared to #853? Is https://github.com/paperless-ngx/paperless-ngx/pull/942/files#diff-c713cc1849ad8b49e415b4b4c4a0c30cfa20c242b18ed42e6997d0aa2c178d35R163 and https://github.com/paperless-ngx/paperless-ngx/pull/942/files#diff-c713cc1849ad8b49e415b4b4c4a0c30cfa20c242b18ed42e6997d0aa2c178d35R164 "the magic"?

File: src/documents/file_handling.py (hunk 1 above)
Comment (stumpylog) [reply src/documents/file_handling.py:164]: It kind of dealt with it, but didn't really, so I no longer do.  It weirdly only checked that for the year, month or day, but not when using the date as a whole.  So if a date could not exist already, there would have been reports of it,

For both `.created` and `.added`, the field definition doesn't include `null=True`, so I don't think it's possible to get these fields to not exist.  They also have a default set.

File: src/documents/tests/test_document_model.py (hunk 2 above)
Comment (stumpylog) [reply src/documents/tests/test_document_model.py:87]: That's an interesting way of thinking about test scenarios.   I like it.  I moved this one down with the other in the case.

File: src/documents/file_handling.py (hunk 1 above)
Comment (lippoliv) [reply src/documents/file_handling.py:164]: Understood. Thought already what the scenario would be, in which added / created do not exist...

**Review by shamoon — APPROVED** [review]

**Pull request merged**